strawberryfields/gbs/data.py

```diff
@@ -0,0 +1,199 @@
+range of input graphs. Each dataset corresponds to a set of samples generated from a fixed graph
+with corresponding values for:
+
+- ``n_mean``: mean number of photons in the GBS device
+- ``n_max``: maximum number of photons allowed in any sample
+-  ``threshold``: flag to indicate whether samples are generated with threshold detection or
+   with photon number resolving detectors.
+- ``n_samples``: total number of samples in dataset
+- ``modes``: number of modes in GBS device or, equivalently, number of nodes in graph
+
+Datasets are available as classes. The following graphs and datasets are provided:
+
+- **Planted graph**: A 30-node graph with a dense 10-node subgraph planted inside
+  :cite:`arrazola2018using`.
+
+  .. image:: ../../_static/planted.png
+        :align: center
+        :width: 300px
+
+  Provided by the :class:`Planted` Dataset with attributes ``n_mean = 6``, ``n_max = 14``,
+  ``threshold = True``, ``n_samples = 50000``, ``modes = 30``.
+
+Loading data
+^^^^^^^^^^^^
+
+We use the :class:`Planted` class as an example to show how to interact with the datasets. Datasets
+can be loaded by running:
+
+>>> data = Planted()
+
+Accessing samples from the dataset is then simple:
+
+>>> i = 3
+>>> sample_i = data[i]
+>>> samples = data[:10]
+
+Datasets also contain metadata relevant to the GBS setup:
+
+>>> data.n_mean
+6
+
+>>> data.n_samples
+50000
+
+Counting photons or clicks in each sample is available using the :meth:`Dataset.counts` method:
+
+>>> data.counts()
+[2, 0, 8, 11, ... , 6]
+
+Code details
+^^^^^^^^^^^^
+
+The :class:`Dataset` class provides the base functionality from which all datasets such as
+:class:`Planted` inherit.
+"""
+# pylint: disable=unnecessary-pass
+from abc import ABCMeta, abstractmethod
+
+import pkg_resources
+import numpy as np
+import scipy
+
+DATA_PATH = pkg_resources.resource_filename("strawberryfields", "gbs/data") + "/"
+
+
+class Dataset(metaclass=ABCMeta):
+    """Base class for loading datasets of pre-generated samples.
+
+    Attributes:
+        n_mean (float): mean number of photons in the GBS device
+        n_max (float): maximum number of photons allowed in any sample. This number is set to
+            limit the computation time, any sample being simulated that exceeds ``n_max`` will be
+            ignored and cause calculation to skip to the next sample.
+        threshold (bool): flag to indicate whether samples are generated with threshold detection
+            (i.e., detectors of zero or some photons) or with photon number resolving detectors.
+        adj (array): adjacency matrix of graph from which samples were generated
+        n_samples (int): total number of samples in dataset
+        modes (int): number of modes in GBS device or, equivalently, number of nodes in graph
+    """
+
+    _count = 0
+
+    @property
+    @abstractmethod
+    def _data_filename(self) -> str:
+        """Name of files stored in ``./data/``. Samples should be provided as a

```

Comment:
```suggestion
        """Base name of files containing the sample data stored in the ``./data/`` directory.
        
        Samples should be provided as a
```

(I like the convention of separating the short summary of the docstring from the remainder of the docstring, I find it makes it easier to skim/parse).